### DISCUSSION TITLE
not ascii characters in Cookies

### DISCUSSION THREAD

**Discussion by kox-solid** [encode/httpx#3240] [Potential Issue]
I need to pass a cookie with non-ascii characters to Client:

```python
from httpx import Client, Cookies

cookies = Cookies()
cookies.set("blablalba", "sprühkopf", "example.com")

with Client(cookies=cookies) as client:
    response = client.get("https://example.com")
```
raise `UnicodeEncodeError: 'ascii' codec can't encode character '\xfc' in position 13: ordinal not in range(128)` in Headers.__setitem__()

Class Headers has a mechanism to switch to "iso-8859-1" encoding for situations like this, but it only works during a session. Class Request creates a new instance of Headers with "ascii" encoding by default, so there are no methods to set a different encoding other than Monkey Patching.

Maybe I'm wrong?

**Comment by SonderZhong** [comment]
You can try the following methods, which may solve your problem.
#3238

**Reply by kox-solid** [reply]
> You can try the following methods, which may solve your problem.
Thanks for your interest, but it can't solve my problem because your method doesn't change the encoding, which is the problem.
